### ISSUE TITLE
 Max. Number of Participants configured in event is taking into account the non-attending registrant

### ISSUE THREAD

**Issue by cristina-p** [twomice/com.joineryhq.groupreg#22]
When the event is set with a Max. Number of Participants and the registered **attendee participants** have reached that max. number minus the number of additional participants the user want to book the event booking advise the following:

"_Please correct the following errors in the form fields below:
There is only enough space left on this event for 1 participant(s).
Only 1 Registrations available_."

It seems that the extension is taking into account the **non-attending registrant** as an attendee participant and doesn't allow to register the additional participants, despite not having reached the max. number of participants.

**Comment by twomice** [comment]
@cristina-p This extension doesn't handle any enforcement of Max Number of Participants, rather that's a feature of CiviCRM itself. It sounds like you will want to edit the relevant "non-attending" role to ensure it's not "counted". You can do that in CiviCRM's "Participant Roles" confirugation:

![counted](https://github.com/twomice/com.joineryhq.groupreg/assets/759449/064204b1-91fb-4b54-b327-8206073e4ef3)

**Comment by cristina-p** [comment]
Many thanks for your prompt response.

I already have this configuration in the Participant Roles. 
Could it be something related to the configuration of the Price Set, on the **Participant Count** field of the Price Option?
![image](https://github.com/twomice/com.joineryhq.groupreg/assets/36039284/25b405c5-24ff-4160-8a98-94bb1a167ad0)

**Comment by twomice** [comment]
Thanks for the extra info @cristina-p . It's true, CiviCRM does use the "participant count" setting on price set fields in its calculation of total participants against the max.

But what you wrote here does raise quesionts for me (additional emphasis mine):

> ... the warning about the maximum number of participants being reached, only appears when the number of spaces left are **exactly** the number of additional participants the user wants to book.

Where you said "exactly", would it also be true to say "equal to or greater than"? Would you mind to check the actual behavior on that and let me know?

**Comment by cristina-p** [comment]
No, when the number of spaces left are greater than the number of additional participants the user wants to book, the extension does allow the registration of the additional participants.
It only doesn't allow it when the number of spaces left are **equal** to the number of additional participants the user wants to book.

**Comment by twomice** [comment]
Right. Where you say,
> It only doesn't allow it when the number of spaces left are **equal** to the number of additional participants the user wants to book.

would it also be true to say 
> It only doesn't allow it when the number of spaces left are **equal to or less than** the number of additional participants the user wants to book.

?

**Comment by cristina-p** [comment]
Yes, that is also true. 
But I just wanted to highlight the fact that when the number of spaces left are **equal** to the number of additional participants the user wants to book, the extension doesn't allow the registration, which is **not** the **behaviour expected**.
When the number of spaces left are **less** than the number of additional participants the user wants to book, the extension doesn't allow the registration either, but this is the **expected behaviour**.

**Comment by twomice** [comment]
Got it, thanks. This is a little tricky, considering the way CiviCRM is doing its calculation. I'll see what I can do.
